pkg/image/source.go

```diff
@@ -46,22 +64,82 @@ func ParseImageSpec(imageSpec string) (Source, string) {
 	return source, strings.TrimPrefix(imageSpec, candidates[0]+"://")
 }
 
-// ParseSource attempts to resolve a concrete image source selection from a user string (e.g. "docker://", "tar://",
-// "podman://", etc.)
+// ParseSource attempts to resolve a concrete image source selection from a user string (e.g. "docker://", "tar://", "podman://", etc.).
 func ParseSource(source string) Source {
 	source = strings.ToLower(source)
 	switch source {
-	case "tarball", "tar", "archive", "docker-archive":
+	case "tarball", "tar", "archive", "docker-archive", "docker-tar", "docker-tarball":
 		return DockerTarballSource
 	case "docker", "docker-daemon", "docker-engine":
 		return DockerDaemonSource
+	case "oci", "oci-directory", "oci-dir":
+		return OciDirectorySource
+	case "oci-tarball", "oci-tar", "oci-archive":
+		return OciTarballSource
 	case "podman":
 		// TODO: implement
 		return UnknownSource
 	}
 	return UnknownSource
 }
 
+// DetectSourceFromPath will distinguish between a oci-layout dir, oci-archive, and a docker-archive.
+func DetectSourceFromPath(imgPath string) (Source, error) {
+	return detectSourceFromPath(afero.NewOsFs(), imgPath)
+}
+
+// detectSourceFromPath will distinguish between a oci-layout dir, oci-archive, and a docker-archive for a given filesystem.
+func detectSourceFromPath(fs afero.Fs, imgPath string) (Source, error) {
```

Comment:
It looks like we're never using the `error` return value. If that's true, I'd suggest removing it.

Reply:
I've added using these errors up to the parent caller